sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/EncryptionCosmosAsyncContainer.java

```diff
@@ -69,6 +69,63 @@
             );
     }
 
+    private Mono<CosmosItemResponse<byte[]>> replaceItemStream(byte[] payload,
+                                                               String itemId,
+                                                               PartitionKey partitionKey,
+                                                               EncryptionItemRequestOptions encryptionItemRequestOptions) {
+        Preconditions.checkNotNull(payload, "payload can't be null");
+
+        // TODO: add diagnostics
+        assert encryptionItemRequestOptions != null && encryptionItemRequestOptions.getEncryptionOptions() != null;
+        payload = EncryptionProcessor.encryptAsync(payload, encryptor,
+            encryptionItemRequestOptions.getEncryptionOptions());
+
+        Mono<CosmosItemResponse<byte[]>> response = container.replaceItem(payload,
+            itemId,
+            partitionKey,
+            encryptionItemRequestOptions);
+
+        return response
+            .subscribeOn(Schedulers.elastic())
+            .publishOn(Schedulers.elastic())
```

Comment:
This publishOn will have no effect on the reactor chain, since subscribeOn will already switch the threads to `Schedulers.elastic()`. Take a look at this example at `subscribeOn` from this article: https://www.woolha.com/tutorials/project-reactor-publishon-vs-subscribeon-difference

If you were using different schedulers, then it would make sense to place both of them here. 

So I guess simpler code will be something like this: 

```
return response
 .map(rsp -> {})
 .subscribeOn(Schedulers.elastic());
```

Reply:
you are right. copy pasted code from some other code I have. will fix shortly.

Thanks.

Reply:
addressed.